### PR TITLE
Use ZeroBuf events for steering and streaming

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: monsteer/plugin/endOfStream.fbs [hunk 1]
```diff
@@ -6,7 +6,5 @@ namespace monsteer.plugin;
 
 table EndOfStream
 {
-endofstream:bool = true;
+  endofstream:bool = true;
```
Comment (hernando) [inline monsteer/plugin/endOfStream.fbs]: This is actually an empty event. The attribute was added a temporary workaround for flat buffer event with not payload not being properly supported.
I told @bilgili to add a comment so we wouldn't forget about removing the attribute, but he was too sure (or maybe eager?) that he would do the change.

File: monsteer/plugin/spikeReport.cpp [hunk 2]
```diff
@@ -120,8 +115,13 @@ float SpikeReport::getEndTime() const
 
 void SpikeReport::writeSpikes( const brion::Spikes& spikes )
 {
-    const zeroeq::FBEvent& event = serializeSpikes( spikes );
-    _publisher->publish( event );
+    SpikesEvent event;
+    for( std::multimap< float, uint32_t >::const_iterator i = spikes.begin();
```
Comment (hernando) [inline monsteer/plugin/spikeReport.cpp]: Use range based for

File: monsteer/plugin/spikeReport.cpp [hunk 3]
```diff
@@ -248,19 +248,20 @@ void SpikeReport::clear( const float startTime, const float endTime )
...
-void SpikeReport::_onSpikes( const zeroeq::FBEvent& event )
+void SpikeReport::_onSpikes( ConstSpikesEventPtr event )
 {
-    LBASSERT( event.getTypeIdentifier() == EVENT_SPIKES );
-    const SpikeMap& spikes = deserializeSpikes( event );
-    _incoming.insert( spikes.begin(), spikes.end( ));
+    for( size_t i = 0; i < event->getSpikes().size( ); ++i )
+    {
+        const Spike& spike = event->getSpikes()[i];
+        _incoming.insert( std::make_pair( spike.getTime(), spike.getCell( )));
```
Comment (hernando) [inline monsteer/plugin/spikeReport.cpp]: Spikes should be sorted at origin, use the insert with hint to insert at the end for performance.

File: monsteer/steering/playbackState.fbs [hunk 4]
```diff
@@ -3,10 +3,14 @@
...
 
+enum State : uint
+{
+  PAUSE
+  PLAY
+}
+
 table PlaybackState
 {
   messageID:string;
```
Comment (hernando) [inline monsteer/steering/playbackState.fbs]: I don't see the messageID being used anywhere and I can't find a future use for keeping it.

File: apps/music_proxy.cpp [hunk 5]
```diff
@@ -205,48 +205,31 @@ class SteeringHandler
...
-
 private:
 
-    void _onStimulusInjection( const zeroeq::FBEvent& event )
+    void _onStimulusInjection( monsteer::steering::ConstStimulusInjectionPtr event )
     {
-        LBASSERT( event.getTypeIdentifier() == monsteer::steering::EVENT_STIMULUSINJECTION )
-        const std::string& json = deserializeJSON( event,
-                                monsteer::steering::SCHEMA_STIMULUSINJECTION );
+        const std::string& json = event->toJSON();
```
Comment (hernando) [inline apps/music_proxy.cpp]: The function toJSON produces a JSON string that includes only the event data or also the event type?
If the event type is part of the JSON string, then we can remove the messageID attribute from the fbs file and adapt the Python code accordingly.

Comment (tribal-tec) [inline apps/music_proxy.cpp]: The JSON includes only event data iirc. I just verified together with @jafyvilla that the steering worked while running the music_proxy with NEST.

Comment (hernando) [inline apps/music_proxy.cpp]: Yesterday I wasn't seeing it clearly. Yes, the current code hasn't really changed anything. The solution that we implemented was to include the message type as a message attribute and this hasn't been modified.

**Pull request merged**